python/pfs/drp/stella/fluxModelInterpolator.py

```diff
@@ -13,6 +13,81 @@
 __all__ = ("FluxModelInterpolator",)
 
 
+class PCACompositor:
+    """Given basis vectors, compute their linear combination.
+
+    Parameters
+    ----------
+    basis : `np.ndarray`
+        Shape (M, N), M being the number of basis vectors
+        and N being the dimension of the vectors.
+
+    mean : `np.ndarray`
+        Shape (N,). This vector is added to linear combinations
+        of basis vectors.
+
+    wcs : `astropy.io.fits.Header`
+        WCS for ``mean`` and for vectors output from ``__call__()``
+    """
+
+    def __init__(self, basis: np.ndarray, mean: np.ndarray, wcs: astropy.io.fits.Header) -> None:
+        self.basis = np.asarray(basis)
+        self.mean = np.asarray(mean)
+        self.wcs = wcs
+
+    def __call__(self, coeff: np.ndarray) -> np.ndarray:
+        r"""Compute a linear combination (+ ``self.mean``).
+
+        This method returns ``mean + \sum_i coeff[..., i] basis[i]``
+
+        Parameters
+        ----------
+        coeff : `np.ndarray`
+            Shape (..., M), M being the number of basis vectors.
+            Coefficients of the linear combination.
+
+        Returns
+        -------
+        sum : `np.ndarray`
+            Shape (..., N), N being the dimension of a basis vector.
+        """
+        coeff = np.asarray(coeff)
+        lenShape = len(coeff.shape)
+        if lenShape > 1:
+            ones = (1,) * (lenShape - 1)
+            mean = self.mean.reshape(ones + self.mean.shape)
```

Comment:
I don't think
```
addedAxes = tuple(range(lenShape - 1))
mean = np.expand_dims(mean, axis=addedAxes)
```
is more readable than what I wrote:
```
ones = (1,) * (lenShape - 1)
mean = self.mean.reshape(ones + self.mean.shape)
```
I like the latter because it is less cryptic.